tests/optimisation_test/recipe_converter.py

```diff
@@ -0,0 +1,61 @@
+import pandas as pd
+import yaml
+from archemist.core.persistence.yaml_handler import YamlHandler
+from nested_lookup import nested_update
+from pathlib import Path
+
+class RecipeUpdate:
+    def __init__(self) -> None:
+        self.placeholder_keys = []
+        self.placeholders =[]
+        super().__init__()
+
+    def find_placeholders(self, data):
+        self._data = data
+        if isinstance(self._data, dict):
+            for key, value in self._data.items():
+                if isinstance(value, str) and '{{' in value and '}}' in value:
```

Comment:
I think it would be more robust if we use regular expression to to parse the string. Check https://docs.python.org/3/library/re.html for more details